renku/ui/service/controllers/utils/project_clone.py

```diff
@@ -33,7 +34,7 @@ def user_project_clone(cache, user_data, project_data):
     project.abs_path.mkdir(parents=True, exist_ok=True)
 
     try:
-        with project.write_lock():
+        with project.write_lock(), renku_project_context(project.abs_path):
```

Comment:
Should we optimize `renku_project_context` to not search for a git repo in this case that we know there is none? This might make sense since it's called in the service for each request (that requires a clone).